provider/packet/environ.go

```diff
@@ -316,6 +366,114 @@ func (e *environ) StartInstance(ctx context.ProviderCallContext, args environs.S
 	}, nil
 }
 
+// supportedInstanceTypes returns the instance types supported by packet.
+func (e *environ) supportedInstanceTypes() ([]instances.InstanceType, error) {
+	plans, _, err := e.packetClient.Plans.List(new(packngo.ListOptions))
```

Comment:
You should probably filter the plans by region and project ID as some of the plans might not be available to certain region/project combinations.

Reply:
That is correct. Here's an example: 
https://github.com/packethost/packngo/blob/master/plans_test.go#L8-L33